storage/client/client.go

```diff
@@ -656,11 +665,17 @@ func (c *StorageClient) Account(ctx context.Context, address staking.Address) (*
 	qf := NewQueryFactory(cid, "" /* no runtime identifier for the consensus layer */)
 
 	a := Account{
-		Allowances: []Allowance{},
+		// Initialize optional fields to empty values to avoid null pointer dereferences
+		// when filling them from the database.
+		Allowances:                  []Allowance{},
+		AddressPreimage:             &AddressPreimage{},
+		DelegationsBalance:          &common.BigInt{},
+		DebondingDelegationsBalance: &common.BigInt{},
```

Comment:
Our docs say: "For efficiency, this field is omitted when listing multiple-accounts."
It's because we need a join and a sum (IIRC) to derive these.

We could have separate types for Account and AccountThatAppearsInAnAccountList, but ... not worth the code duplication IMO.

Reply:
Er but since this is the `Account` endpoint (not `Accounts`), the `AccountQuery` only fetches one account and so it will never be null due to the `COALESCE`. Can we remove these two lines here and simply treat `DelegationsBalance` and `DebondingDelegationsBalance` like the other fields, eg 
```
                a.Debonding, err = c.numericToBigInt(ctx, &debondingNum)
		if err != nil {
			return nil, apiCommon.ErrStorageError
		}
		a.DelegationsBalance, err = c.numericToBigInt(ctx, &delegationsBalanceNum)
		if err != nil {
			return nil, apiCommon.ErrStorageError
		}
		a.DebondingDelegationsBalance, err = c.numericToBigInt(ctx, &debondingDelegationsBalanceNum)
		if err != nil {
			return nil, apiCommon.ErrStorageError
		}
```

Reply:
I see. Unfortunately not, because `numericToBigInt` returns a `BigInt`, so we need to assign it to something of type `BigInt`. 
`a.DelegationsBalance` is `*BigInt` (enforced by the yaml), so we need to assign with a `*` to make the type checker happy:
```
*a.DelegationsBalance, err = c.numericToBigInt(ctx, &delegationsBalanceNum)
``` 
and dereferencing `a.DelegationsBalance` when it's `nil` (which it will be without the line that started this comment thread) leads to having a bad time.